apps/nestjs-api/src/reminder-emails/reminder-emails.service.ts

```diff
@@ -0,0 +1,279 @@
+import { Injectable } from '@nestjs/common'
+import { ConfigService } from '@nestjs/config'
+import {
+  ConnectProfileStatus,
+  MentorshipMatchStatus,
+  RediLocation,
+  UserType,
+} from '@talent-connect/common-types'
+import * as AWS from 'aws-sdk'
+import * as jsforce from 'jsforce'
+import { ConMentorshipMatchesService } from '../con-mentorship-matches/con-mentorship-matches.service'
+import { ConProfilesService } from '../con-profiles/con-profiles.service'
+import { SfApiEmailTemplatesService } from '../salesforce-api/sf-api-email-templates.service'
+
+const isProductionOrDemonstration = () =>
+  ['production', 'demonstration', 'staging'].includes(process.env.NODE_ENV)
+
+@Injectable()
+export class ReminderEmailsService {
+  private ses: AWS.SES
+
+  constructor(
+    private readonly emailTemplatesService: SfApiEmailTemplatesService,
+    private readonly conProfilesService: ConProfilesService,
+    private readonly conMentorshipMatchesService: ConMentorshipMatchesService,
+    private readonly config: ConfigService
+  ) {
+    this.ses = new AWS.SES({
+      accessKeyId: this.config.get('NX_EMAILER_AWS_ACCESS_KEY'),
+      secretAccessKey: this.config.get('NX_EMAILER_AWS_SECRET_KEY'),
+      region: this.config.get('NX_EMAILER_AWS_REGION'),
+    })
+  }
+
+  async getDraftingConProfiles({ userType }: { userType: UserType }) {
+    const createdDate = new Date()
+    createdDate.setDate(createdDate.getDate() - 14)
+
+    return this.conProfilesService.findAll({
+      'RecordType.DeveloperName': userType,
+      Profile_Status__c: ConnectProfileStatus.DRAFTING_PROFILE,
+      Created_Date_Without_Time__c: {
+        $eq: jsforce.SfDate.toDateLiteral(createdDate),
+      },
+    })
+  }
+
+  async getApprovedMenteesByDaysAndLocation({
+    daysAgo,
+  }: {
+    daysAgo: '7d' | '14d'
+  }) {
+    const approvedDate = new Date()
+    approvedDate.setDate(approvedDate.getDate() - parseInt(daysAgo)) // Interestingly, parseInt is able to parse 7d and 14d to 7 and 14 respectively
+
+    return this.conProfilesService.findAll({
+      'RecordType.DeveloperName': UserType.MENTEE,
+      Profile_Status__c: ConnectProfileStatus.APPROVED,
+      Profile_First_Approved_At__c: {
+        $eq: jsforce.SfDate.toDateLiteral(approvedDate),
+      },
+    })
+  }
```

Comment:
Do we make sure we get the profiles that have no mentor applications?

Reply:
It's even better now: `getApprovedMenteesWithNoMentorApplicationsByDaysAndLocation`

Reply:
Thanks for the great catch @katamatata. Otherwise this would bug mentees with already mentors.

Reply:
Why by Location?